### PR TITLE
added new tools

### PR DESCRIPTION
**Pull request by qgao-hd**
1) intensity-based image registration; 2) landmark registration using least squares.

### CODE REVIEW

**Comment by bgruening** [comment]
@qgao-hd please do not merge before the test has passed. I will have a look and install the tools this week, ok?

**Comment by qgao-hd** [comment]
Hi @bgruening , I just created a PR (#361) for updating imaging.yaml to add these two tools. The checks failed because the new tools are not in the toolshed. Does it mean that I can not update the file "imaging.yaml" until the new tools are (automatically) integrated? What shall I do wrt the PR #361?

**Comment by qgao-hd** [comment]
> @qgao-hd please do not merge before the test has passed. I will have a look and install the tools this week, ok?

I did the merge after all checks had passed. 

Thanks~

File: tools/landmark_registration_ls/landmark_registration_ls.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <requirement type="package" version="1.15.2">numpy</requirement>
+    </requirements>
+    <command><![CDATA[ 
+         python '$__tool_directory__/landmark_registration_ls.py'
+         '$fn_pts1'
+         '$fn_pts2'
+         '$fn_tmat'
+]]></command>
+    <inputs>
+          <param name="fn_pts1" type="data" format="tabular" label="Path to tab-separated file with src points" />
```
Comment (bgruening) [inline tools/landmark_registration_ls/landmark_registration_ls.xml:15]: It's not a path. Maybe this label can be improved here

File: tools/landmark_registration_ls/landmark_registration_ls.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          <param name="fn_pts2" type="data" format="tabular" label="Path to tab-separated file with dst points" />
+    </inputs>
+    <outputs>
+       <data format="tabular" name="fn_tmat" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="fn_pts1" value="points1.tsv"/>
+            <param name="fn_pts2" value="points2.tsv"/>
+            <output name="fn_tmat" value="tmat.tsv" ftype="tabular" compare="diff" lines_diff="6"/>
```
Comment (bgruening) [inline tools/landmark_registration_ls/landmark_registration_ls.xml:25]: If you have diff lines here with 6 but your file has only 3 lines you are basically testing nothing.
What was the problem here? Rounding differences? Maybe you can use assert based tests?

**Comment by bgruening** [comment]
> Hi @bgruening , I just created a PR (#361) for updating imaging.yaml to add these two tools. The checks failed because the new tools are not in the toolshed. Does it mean that I can not update the file "imaging.yaml" until the new tools are (automatically) integrated? What shall I do wrt the PR #361?

Yes this is correct. I can deal with this as soon as all tools are in the toolshed.

File: tools/landmark_registration_ls/landmark_registration_ls.xml (hunk 1 above)
Comment (qgao-hd) [reply tools/landmark_registration_ls/landmark_registration_ls.xml:15]: I did so to make the description/label consistent with existing tools (eg, landmark_registration). 

I agree with you that this sounds a little bit strange. Maybe we fix such issues (for all tools) later?

File: tools/landmark_registration_ls/landmark_registration_ls.xml (hunk 2 above)
Comment (qgao-hd) [reply tools/landmark_registration_ls/landmark_registration_ls.xml:25]: Yes, there are always numerical errors (finite precision of computations).

I copied this line from the existing xml of the tool landmark_registration. I'd mentioned in the email that the automatic testing is very limited. You said it mainly/only tests the wrapper, parameters, outputs...

Functional tests have been done off-line.

Comment (bgruening) [reply tools/landmark_registration_ls/landmark_registration_ls.xml:25]: > I copied this line from the existing xml of the tool landmark_registration. I'd mentioned in the email that automatic testing is very limited. You said it mainly/only tests the wrapper, parameters, outputs...

Yes, but this test is really doing nothing. So if for some reason the file is empty is test would also pass. Have a look at https://docs.galaxyproject.org/en/latest/dev/schema.html#id79 we can even use regex is such tests to adjust for numerical instability.

File: tools/landmark_registration_ls/landmark_registration_ls.xml (hunk 1 above)
Comment (bgruening) [reply tools/landmark_registration_ls/landmark_registration_ls.xml:15]: Yes that would be great!

**Pull request merged**